src/udev/net/link-config.c

```diff
@@ -269,7 +269,11 @@ int link_load_one(LinkConfigContext *ctx, const char *filename) {
         for (i = 0; i < ELEMENTSOF(config->features); i++)
                 config->features[i] = -1;
 
-        dropin_dirname = strjoina(basename(filename), ".d");
+        r = path_extract_filename(filename, &file_basename);
+        if (r < 0)
+                return log_error_errno(r, "Failed to extract basename of '%s', ignoring: %m", filename);
```

Comment:
drop the ", ignoring" here, you are not ignroing this failure here, but failing the operation.

Reply:
Yeah, I think in general I was a bit confused about what "ignoring" meant - I thought it meant "ignoring the drop-in", not "ignoring the error and continuing".  I'll fix this in all my log messages.